Breaks freeimage-unstable-2021-11-01 build with:
```
Source/Metadata/XTIFF.cpp: In function 'BOOL tiff_write_exif_tags(TIFF*, TagLib::MDMODEL, FIBITMAP*)':
Source/Metadata/XTIFF.cpp:752:38: error: '_TIFFDataSize' was not declared in this scope; did you mean 'TIFFDataType'?
  752 |                         if((unsigned)_TIFFDataSize(tif_tag_type) != FreeImage_TagDataWidth(tag_type)) {
      |                                      ^~~~~~~~~~~~~
      |                                      TIFFDataType
```

freeimage is using a non-public libtiff API that was removed and replaced with a public API. This had [already been reported on the freeimage forum](https://sourceforge.net/p/freeimage/discussion/36109/thread/2018fdc6e7/) and the [libtiff mailing list](https://www.asmail.be/msg0055228425.html). I'll try applying that patch.

--- a/pkgs/development/libraries/freeimage/default.nix
+++ b/pkgs/development/libraries/freeimage/default.nix
@@ -15,8 +15,13 @@ stdenv.mkDerivation {
   sourceRoot = "svn-r1900/FreeImage/trunk";
 
   # Ensure that the bundled libraries are not used at all
-  prePatch = "rm -rf Source/Lib* Source/OpenEXR Source/ZLib";
-  patches = [ ./unbundle.diff ./libtiff-4.4.0.diff ];
+  prePatch = ''
+    rm -rf Source/Lib* Source/OpenEXR Source/ZLib
+  '';
+  patches = [
+    ./unbundle.diff
+    ./libtiff-4.4.0.diff
+  ];
 
   postPatch = ''
     # To support cross compilation, use the correct `pkg-config`.

--- a/pkgs/development/libraries/freeimage/default.nix
+++ b/pkgs/development/libraries/freeimage/default.nix
@@ -16,7 +16,7 @@ stdenv.mkDerivation {
 
   # Ensure that the bundled libraries are not used at all
   prePatch = "rm -rf Source/Lib* Source/OpenEXR Source/ZLib";
-  patches = [ ./unbundle.diff ];
+  patches = [ ./unbundle.diff ./libtiff-4.4.0.diff ];
 
   postPatch = ''
     # To support cross compilation, use the correct `pkg-config`.

--- a/pkgs/development/libraries/freeimage/libtiff-4.4.0.diff
+++ b/pkgs/development/libraries/freeimage/libtiff-4.4.0.diff
@@ -1,0 +1,15 @@
+Fix build with libtiff 4.4.0 by not using a private libtiff API.
+Patch by Kurt Schwehr: https://sourceforge.net/p/freeimage/discussion/36109/thread/2018fdc6e7/
+
+diff -ru a/Source/Metadata/XTIFF.cpp b/Source/Metadata/XTIFF.cpp
+--- a/Source/Metadata/XTIFF.cpp
++++ b/Source/Metadata/XTIFF.cpp
+@@ -749,7 +749,7 @@
+ 				continue;
+ 			}
+ 			// type of storage may differ (e.g. rationnal array vs float array type)
+-			if((unsigned)_TIFFDataSize(tif_tag_type) != FreeImage_TagDataWidth(tag_type)) {
++			if((unsigned)TIFFFieldSetGetSize(fld) != FreeImage_TagDataWidth(tag_type)) {
+ 				// skip tag or _TIFFmemcpy will fail
+ 				continue;
+ 			}

--- a/pkgs/development/python-modules/pillow/default.nix
+++ b/pkgs/development/python-modules/pillow/default.nix
@@ -1,5 +1,6 @@
 { lib
 , stdenv
+, fetchpatch
 , buildPythonPackage
 , pythonOlder
 , fetchPypi
@@ -21,6 +22,14 @@ import ./generic.nix (rec {
     inherit version;
     sha256 = "f401ed2bbb155e1ade150ccc63db1a4f6c1909d3d378f7d1235a44e90d75fb97";
   };
+
+  patches = [
+    # Fix failing test with libtiff 4.4.0
+    (fetchpatch {
+      url = "https://github.com/python-pillow/Pillow/commit/40a918d274182b7d7c063d7797fb77d967982c4a.patch";
+      sha256 = "sha256-f8m3Xt3V3pHggK1JEc2tnPmrTVPFjfV4YJqwE1KM1pA=";
+    })
+  ];
 
   passthru.tests = {
     inherit imageio matplotlib pilkit pydicom reportlab;

--- a/pkgs/development/python-modules/pillow/generic.nix
+++ b/pkgs/development/python-modules/pillow/generic.nix
@@ -2,6 +2,7 @@
 , version
 , disabled
 , src
+, patches ? []
 , meta
 , passthru ? {}
 , ...
@@ -10,7 +11,7 @@
 with args;
 
 buildPythonPackage rec {
-  inherit pname version src meta passthru;
+  inherit pname version src meta passthru patches;
 
   # Disable imagefont tests, because they don't work well with infinality:
   # https://github.com/python-pillow/Pillow/issues/1259

